src/librustdoc/html/render/mod.rs

```diff
@@ -2101,112 +2140,121 @@ pub(crate) fn sidebar_render_assoc_items(
                     id_map.derive(get_id_for_impl(&it.inner_impl().for_, Some(trait_), cx));
 
                 let i_display = format!("{:#}", trait_.print(cx));
-                let out = Escape(&i_display);
                 let prefix = match it.inner_impl().polarity {
                     ty::ImplPolarity::Positive | ty::ImplPolarity::Reservation => "",
                     ty::ImplPolarity::Negative => "!",
                 };
-                let generated = format!("<a href=\"#{}\">{}{}</a>", encoded, prefix, out);
+                let generated = SideBarAssocItem { encoded, prefix, content: i_display };
                 if links.insert(generated.clone()) { Some(generated) } else { None }
             })
-            .collect::<Vec<String>>();
+            .collect::<Vec<_>>();
```

Comment:
Why remove the type from the turbofish here? It seems better to write out the new type (SideBarAssocItem), so the reader knows what type `ret` is.